### PR TITLE
Patch cuffmerge for python3

### PR DESCRIPTION
**Pull request by yp**
Cuffmerge uses some methods that are not available on python3. This commit patches on-the-fly the script replacing them with the corresponding python3 constructs.

### CODE REVIEW

File: recipes/cufflinks/build.sh [hunk 1]
```diff
@@ -5,6 +5,36 @@ mkdir -p $PREFIX/bin
...
+         line = "@SQ\tSN:%s\tLN:\t%d" % (chrom, limits[1])
+@@ -523,7 +523,7 @@
+         start_time = datetime.now()
+         prepare_output_dir()
+
+-        run_log = open(logging_dir + "run.log", "w", 0)
++        run_log = open(logging_dir + "run.log", "w")
+         run_cmd = " ".join(argv)
+         print(run_cmd, file=run_log)
+EOF
```
Comment (johanneskoester) [inline recipes/cufflinks/build.sh]: Thank you, that's great! Can you move this into a separate patch file and add it via the meta.yaml (described [here](http://conda.pydata.org/docs/building/meta-yaml.html#patches))?
Apart from that, do you want to join the team?

Comment (yp) [inline recipes/cufflinks/build.sh]: Thanks for you work on bioconda, instead!
The patch is needed only if the interpreter is python3 (maybe it works on python2 too, but maybe it is better to not modify the original code here), while the meta.yaml file would apply it in any case, wouldn't it?
I don't think I will be able to make big contributions, but I would be glad.

Comment (johanneskoester) [inline recipes/cufflinks/build.sh]: You can add a ``# [py3k]`` to the patch line in the meta.yaml. Then the patch will be only applied unless the recipe is built on Python 3, i.e.:

```
source:
  patches:
    - py3.patch # [py3k]
```

Comment (yp) [inline recipes/cufflinks/build.sh]: Thanks, I did not know!
I updated the pull request accordingly.
